kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriberServiceImpl.java

```diff
@@ -584,30 +599,45 @@ public ChildImportRejection updateRchChildSubscriber(Long msisdn, MctsChild chil
             }
         }
 
-        Subscriber subscriber = finalSubscription != null ? finalSubscription.getSubscriber() : null;
-        Subscription pregnancySubscription = subscriptionService.getActiveSubscription(subscriber,
-                SubscriptionPackType.PREGNANCY);
-        if (pregnancySubscription != null) {
-            subscriptionService.deactivateSubscription(pregnancySubscription, DeactivationReason.LIVE_BIRTH);
-        }
-
-        Boolean death = (Boolean) record.get(KilkariConstants.DEATH);
-        if ((death != null) && death) {
-            subscriptionService.deactivateSubscription(finalSubscription, DeactivationReason.CHILD_DEATH);
-        }
+        liveBirthChildDeathCheck(finalSubscription, record);
 
         return childRejectionRch(convertMapToRchChild(record), true, null, action);
     }
 
-    public Subscription updateOrCreateSubscription(Subscriber subscriber, Subscription subscription, DateTime dateTime, SubscriptionPack pack, Language language, Circle circle, SubscriptionOrigin origin) {
+    public Subscription updateOrCreateSubscription(Subscriber subscriber, Subscription subscription, DateTime dateTime, SubscriptionPack pack, Language language, Circle circle, SubscriptionOrigin origin, Boolean greaterCaseNo) { // NO CHECKSTYLE Cyclomatic Complexity
+        Subscription deactivatedSubscripion = subscriptionService.getLatestDeactivatedSubscription(subscriber, pack.getType());
         if (subscription != null && (subscription.getStatus() == SubscriptionStatus.ACTIVE || subscription.getStatus() == SubscriptionStatus.PENDING_ACTIVATION)) {
```

Comment:
Better to have constant like "ACTIVE" on the left -- also better to assign getStatus() to a variable and use it do the comaprision